### PR TITLE
fix(frontend): Search results should reset after reopening the "Add course" pop-up window tm-315

### PR DESCRIPTION
**Pull request by GEOFARL**
Search results reset after reopening the "Add course" pop-up window, already added course is not shown with the active "Add" button

### CODE REVIEW

File: apps/frontend/src/modules/courses/slices/courses.slice.ts [hunk 1]
```diff
@@ -43,7 +44,28 @@ const { actions, name, reducer } = createSlice({
...
+				return !action.payload.some(
+					(payload) =>
+						payload.vendorCourseId === course.vendorCourseId &&
+						payload.vendorId === course.vendor.id,
+				);
+			});
+
+			state.searchedCourses = state.searchedCourses.filter((course) => {
+				return !action.payload.some(
+					(payload) =>
```
Comment (Oleksandra-Masiuk) [inline apps/frontend/src/modules/courses/slices/courses.slice.ts:62]: B27

**Review by Vfivew — APPROVED** [review]

**Review by Ann-Maksay — APPROVED** [review]

**Review by Sytnikovakatya — APPROVED** [review]

File: apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx [hunk 2]
```diff
@@ -104,8 +107,14 @@ const AddCourseModal: React.FC<Properties> = ({ onClose }: Properties) => {
 
 	const hasCourses = courses.length > EMPTY_ARRAY_LENGTH;
 
+	const extendedOnClose = useCallback((): void => {
+		dispatch(courseActions.removeAlreadyAddedCourses(addedCourses));
+		setAddedCourses([]);
+		onClose();
+	}, [dispatch, addedCourses, onClose]);
```
Comment (tarmiel) [inline apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:114]: I think the main point of your task wasn't to remove added courses, but to reset searched courses to not see them when we reopen modal. We already have lodic to disable add button

Comment (GEOFARL) [reply apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:114]: I need to remeber somehow which courses were added from that modal window and when we close it just remove them from it, so when we reopen it later, they are not displayed again and we won't have ability to try to add already added course item.

image

File: apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx [hunk 3]
```diff
@@ -104,8 +107,14 @@ const AddCourseModal: React.FC<Properties> = ({ onClose }: Properties) => {
 
 	const hasCourses = courses.length > EMPTY_ARRAY_LENGTH;
 
+	const extendedOnClose = useCallback((): void => {
```
Comment (what1s1ove) [inline apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:110]: the name for the handler is not right

File: apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx [hunk 4]
```diff
@@ -51,6 +53,7 @@ const AddCourseModal: React.FC<Properties> = ({ onClose }: Properties) => {
 	const handleAddCourse = useCallback(
 		(payload: AddCourseRequestDto) => {
 			void dispatch(userCourseActions.add(payload));
+			setAddedCourses((previous) => [...previous, payload]);
```
Comment (what1s1ove) [inline apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:56]: why we can not handle it just by clearing state on component unmount?

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx (hunk 4 above)
Comment (GEOFARL) [reply apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:56]: great point

Comment (GEOFARL) [reply apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:56]: I guess we probably then will be making more requests to openAPI which is costly

**Review by fshabanov — APPROVED** [review]

**Review by Vfivew — APPROVED** [review]

File: apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx [hunk 5]
```diff
@@ -104,8 +104,13 @@ const AddCourseModal: React.FC<Properties> = ({ onClose }: Properties) => {
 
 	const hasCourses = courses.length > EMPTY_ARRAY_LENGTH;
 
+	const handleOnClose = useCallback((): void => {
```
Comment (fshabanov) [inline apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx:107]:
```suggestion
	const handleClose = useCallback((): void => {
```

**Review by fshabanov — CHANGES_REQUESTED** [review]
Fix CI issues as well

**Review by YourTeaChan — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**